django/contrib/auth/backends.py

```diff
@@ -90,9 +94,12 @@ def has_module_perms(self, user_obj, app_label):
     def get_user(self, user_id):
         UserModel = get_user_model()
         try:
-            return UserModel._default_manager.get(pk=user_id)
+            user = UserModel._default_manager.get(pk=user_id)
         except UserModel.DoesNotExist:
             return None
+        if user and (getattr(user, 'is_active', None) or self.allow_inactive_users):
```

Comment:
I tend to think of the user doesn't have an `is_active` attribute, it should be accepted regardless of `allow_inactive_users`. Should make the behavior clear in the docs.